media/js/devreg/submit.js

```diff
@@ -105,6 +105,8 @@
         return $('#id_free_platforms option:selected').length ||
             $('#id_paid_platforms option[value="paid-firefoxos"]:selected').length ||
             $('#id_paid_platforms option[value="paid-android-mobile"]:selected').length ||
+            $('#id_paid_platforms option[value="paid-android-tablet"]:selected').length ||
+            $('#id_paid_platforms option[value="paid-desktop"]:selected').length ||
```

Comment:
I think you can replace all of these lines to match the first line (as it is for free) now so that rather than checking for all the specfic paid values you look for if any paid platforms are selected.

e.g:

`#id_paid_platforms option:selected`